zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -194,11 +198,27 @@ public static Either<ProblemDetail, UpdateJobRequest> toJobUpdateRequest(
                     updateRequest.getChangeset().getTimeout())));
   }
 
-  public static Either<ProblemDetail, AuthorizationAssignRequest> toAuthorizationAssignRequest(
-      final AuthorizationAssignRequest authorizationAssignRequest) {
+  public static Either<ProblemDetail, PatchAuthorizationRequest> toAuthorizationAssignRequest(
+      final long ownerKey, final AuthorizationPatchRequest authorizationPatchRequest) {
     return getResult(
-        validateAuthorizationAssignRequest(authorizationAssignRequest),
-        () -> authorizationAssignRequest);
+        validateAuthorizationAssignRequest(authorizationPatchRequest),
+        () -> {
+          final Map<PermissionType, List<String>> permissions = new HashMap<>();
+          authorizationPatchRequest
+              .getPermissions()
+              .forEach(
+                  permission -> {
+                    permissions.put(
+                        PermissionType.valueOf(permission.getPermissionType().name()),
+                        permission.getResourceIds());
+                  });
+
+          return new PatchAuthorizationRequest(
+              ownerKey,
+              PermissionAction.valueOf(authorizationPatchRequest.getAction().name()),
```

Comment:
🔧 I'm not fond of this naming here, PermissionAction implies to me that its something to do direction with the permission, really this is the action of the request. I think this could be generated and therefore out of our hands but if not I'd really prefer something like `AuthorizationRequestAction` so its clear that the action relates to the request as a whole

Reply:
In my mind this action does have something to do directly to do with the permissions inside of the requests. It refers to whether the permissions in the requests are added, or removed.

The way the request works is we patch an Authorization by adding or removing permissions.

Reply:
I see, fair point, I don't feel strongly enough about this to push for any change as you've given me a valid viewpoint :)

Reply:
Agreed :)